crates/bifrost/src/providers/replicated_loglet/network.rs

```diff
@@ -58,10 +107,174 @@ impl RequestPump {
+
+        if !loglet.is_sequencer_local() {
+            return_error_status!(reciprocal, SequencerStatus::NotSequencer);
+        }
+
+        let global_tail = loglet.known_global_tail();
+
+        let loglet_commit = match loglet.enqueue_batch(append.payloads).await {
+            Ok(loglet_commit) => loglet_commit,
+            Err(err) => {
+                return_error_status!(reciprocal, SequencerStatus::from(err), global_tail);
+            }
+        };
+
+        let task = WaitAppendedTask {
+            reciprocal,
+            loglet_commit,
+            global_tail: global_tail.clone(),
+        };
+
+        let _ = task_center().spawn_child(TaskKind::Disposable, "wait-appended", None, task.run());
+    }
+
+    async fn get_loglet<T: TransportConnect>(
+        &self,
+        provider: &ReplicatedLogletProvider<T>,
+        header: &CommonRequestHeader,
+    ) -> Result<Arc<ReplicatedLoglet<T>>, SequencerStatus> {
+        let loglet = match provider.get_active_loglet(header.log_id, header.segment_index) {
+            Some(loglet) => loglet,
+            None => {
+                let logs = self.metadata.logs();
+                let chain = logs
+                    .chain(&header.log_id)
+                    .ok_or(SequencerStatus::UnknownLoglet)?;
+
+                let segment = chain
+                    .iter()
+                    .rev()
+                    .find(|segment| segment.index() == header.segment_index)
+                    .ok_or(SequencerStatus::UnknownLoglet)?;
+
+                provider
+                    .get_or_create_loglet(
+                        header.log_id,
+                        header.segment_index,
+                        &segment.config.params,
+                    )
+                    .map_err(SequencerStatus::from)?
+            }
+        };
+
+        Ok(loglet)
+    }
+}
+
+impl From<OperationError> for SequencerStatus {
+    fn from(value: OperationError) -> Self {
+        match value {
+            OperationError::Shutdown(_) => SequencerStatus::Shutdown,
+            OperationError::Other(err) => Self::Error {
+                retryable: err.retryable(),
+                message: err.to_string(),
+            },
+        }
+    }
+}
+
+impl From<ReplicatedLogletError> for SequencerStatus {
+    fn from(value: ReplicatedLogletError) -> Self {
+        Self::Error {
+            retryable: value.retryable(),
+            message: value.to_string(),
+        }
+    }
+}
+
+struct WaitAppendedTask {
+    loglet_commit: LogletCommit,
+    reciprocal: Reciprocal,
+    global_tail: TailOffsetWatch,
+}
+
+impl WaitAppendedTask {
+    async fn run(self) -> anyhow::Result<()> {
+        let mut cancelled = std::pin::pin!(cancellation_watcher());
+
+        let result = tokio::select! {
+            _ = &mut cancelled => {
+                // we could try to respond to the request still
+                // but it's probably too late.
+                anyhow::bail!(ShutdownError);
```

Comment:
Disposable tasks will already be aborted on cancel so you don't need to watch cancellation here.